### PR TITLE
Convert NuGetGallery.Services to SDK-style project

### PR DESCRIPTION
**Pull request by drewgillies**
Addresses in part https://github.com/NuGet/Engineering/issues/4405

This change does the following:
- Converts `NuGetGallery.Services` to an SDK-style project.
- Removes dependencies on Azure Cloud type `RoleEnvironment` - every time we were accessing this type, it threw (or `IsAvailable` returned null), which means that the code paths that remain are in fact that only paths that have been executing for some time.
- Removes irrelevant tests, and refactored tests to no longer use a cloud setting property which is redundant per above.
- Removes `AssemblyInfo.cs` as this is no longer part of the pattern, and as a result have moved the `InternalsVisibleTo` attribute into the `csproj`.

This change is the foundation for adding a `net6` target.

Testing: Passes UTs/CI, have deployed to DEV and successfully run numerous scenarios (package views, authentication, uploads, downloads, statistics). Telemetry shows no exceptions.

### CODE REVIEW

File: src/NuGetGallery.Services/NuGetGallery.Services.csproj [hunk 1]
```diff
@@ -1,297 +1,44 @@
...
-  <ItemGroup>
-    <ProjectReference Include="..\NuGet.Services.Entities\NuGet.Services.Entities.csproj">
-      <Project>{6262f4fc-29be-4226-b676-db391c89d396}</Project>
-      <Name>NuGet.Services.Entities</Name>
-    </ProjectReference>
-    <ProjectReference Include="..\NuGetGallery.Core\NuGetGallery.Core.csproj">
-      <Project>{097b2cdd-9623-4c34-93c2-d373d51f5b4e}</Project>
-      <Name>NuGetGallery.Core</Name>
-    </ProjectReference>
+    <ProjectReference Include="..\NuGet.Services.Entities\NuGet.Services.Entities.csproj" />
```
Comment (joelverhagen) [inline src/NuGetGallery.Services/NuGetGallery.Services.csproj:28]: This project reference is now implicit via `NuGetGallery.Core.csproj` right?

**Review by joelverhagen — DISMISSED** [review]
🥇 awesome!!!

File: src/NuGetGallery.Services/Configuration/ConfigurationService.cs [hunk 2]
```diff
@@ -168,21 +167,8 @@ public async Task<string> ReadSettingAsync(string settingName)
...
-            }
-            else if (string.IsNullOrEmpty(value))
-            {
-                var cstr = GetConnectionString(settingName);
-                value = cstr != null ? cstr.ConnectionString : GetAppSetting(settingName);
-            }
-
-            return value;
+            var cstr = GetConnectionString(settingName);
+            return cstr != null ? cstr.ConnectionString : GetAppSetting(settingName); ;
```
Comment (shishirx34) [inline src/NuGetGallery.Services/Configuration/ConfigurationService.cs:171]: nit: extra semi-colon

nit not important, can also:
```cs
return cstr?.ConnectionString ?? GetAppSetting(settingName);
```

**Review by shishirx34 — DISMISSED** [review]

File: src/NuGetGallery.Services/NuGetGallery.Services.csproj (hunk 1 above)
Comment (drewgillies) [reply src/NuGetGallery.Services/NuGetGallery.Services.csproj]: Great catch--thanks! I've removed this reference in the next commit.

File: src/NuGetGallery.Services/Configuration/ConfigurationService.cs (hunk 2 above)
Comment (drewgillies) [reply src/NuGetGallery.Services/Configuration/ConfigurationService.cs]: Thanks--I've fixed this in the next commit.

**Review by joelverhagen — APPROVED** [review]

**Pull request merged**